tasks/GenericRawToMotion2D.cpp

```diff
@@ -40,19 +40,58 @@ bool GenericRawToMotion2D::startHook()
 void GenericRawToMotion2D::updateHook()
 {
     RawCommand rcmd;
-    base::commands::Motion2D  mcmd;
-    if(_raw_command.read(rcmd) == RTT::NewData){
-        int trans_axis = _translation_axis.get().at(0);
-        int trans_subaxis = _translation_axis.get().at(1);
-
-        int rot_axis = _rotation_axis.get().at(0);
-        int rot_subaxis = _rotation_axis.get().at(1);
-        double trans_raw = rcmd.axisValue[trans_axis][trans_subaxis];
-        double rot_raw   = rcmd.axisValue[rot_axis][rot_subaxis];
-
-        mcmd.translation = fabs(trans_raw) < _translation_axis_deadzone ? 0.0 : trans_raw * _maxSpeed;
-        double w = (trans_raw < 0.0) ? rot_raw * _maxRotationSpeed : - rot_raw * _maxRotationSpeed;
-        mcmd.rotation = fabs(rot_raw) < _rotation_axis_deadzone ? 0.0 : w;
+    if (_raw_command.read(rcmd) == RTT::NewData) {
+
+        double trans_raw = rcmd.axisValue[_translation_axis.get().at(0)][_translation_axis.get().at(1)];
+        double rot_raw = rcmd.axisValue[_rotation_axis.get().at(0)][_rotation_axis.get().at(1)];
+
+        // just use input values when they exceed deadzone
+        trans_raw = fabs(trans_raw) < _translation_axis_deadzone ? 0.0 : trans_raw;
+        rot_raw = fabs(rot_raw) < _rotation_axis_deadzone ? 0.0 : rot_raw;
+
+
+        if(!_acceleration_mode){
+            mcmd.translation = trans_raw * _maxSpeed;
+            mcmd.rotation = (trans_raw < 0.0) ? rot_raw * _maxRotationSpeed : - rot_raw * _maxRotationSpeed;
+            _motion_command.write(mcmd);
+            return;
```

Comment:
The buttons are not evaluated in case of the acceleration mode. Is this intentional?